flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/serializer/AvroSchemaConvertor.java

```diff
@@ -200,4 +241,204 @@ private static DataType handleLogicalTypeSchema(Schema fieldSchema) {
         }
         return null;
     }
+
+    /**
+     * Converts Flink SQL {@link org.apache.flink.table.types.logical.LogicalType} (can be nested)
+     * into an Avro schema.
+     *
+     * <p>Use "org.apache.flink.avro.generated.record" as the type name.
+     *
+     * @param schema the schema type, usually it should be the top level record type, e.g. not a
+     *     nested type
+     * @return Avro's {@link Schema} matching this logical type.
+     */
+    public static Schema convertToSchema(org.apache.flink.table.types.logical.LogicalType schema) {
+        return convertToSchema(schema, "org.apache.flink.avro.generated.record");
+    }
+
+    /**
+     * Converts Flink SQL {@link org.apache.flink.table.types.logical.LogicalType} (can be nested)
+     * into an Avro schema.
+     *
+     * <p>The "{rowName}_" is used as the nested row type name prefix in order to generate the right
+     * schema. Nested record type that only differs with type name is still compatible.
+     *
+     * @param logicalType logical type
+     * @param rowName the record name
+     * @return Avro's {@link Schema} matching this logical type.
+     */
+    public static Schema convertToSchema(
+            org.apache.flink.table.types.logical.LogicalType logicalType, String rowName) {
+        int precision;
+        boolean nullable = logicalType.isNullable();
+        switch (logicalType.getTypeRoot()) {
+            case NULL:
+                return SchemaBuilder.builder().nullType();
+            case BOOLEAN:
+                Schema bool = SchemaBuilder.builder().booleanType();
+                return nullable ? nullableSchema(bool) : bool;
+            case TINYINT:
+            case SMALLINT:
+            case INTEGER:
+                Schema integer = SchemaBuilder.builder().intType();
+                return nullable ? nullableSchema(integer) : integer;
+            case BIGINT:
+                Schema bigint = SchemaBuilder.builder().longType();
+                return nullable ? nullableSchema(bigint) : bigint;
+            case FLOAT:
+                Schema f = SchemaBuilder.builder().floatType();
+                return nullable ? nullableSchema(f) : f;
+            case DOUBLE:
+                Schema d = SchemaBuilder.builder().doubleType();
+                return nullable ? nullableSchema(d) : d;
```

Comment:
Lets use better names. `f` `d` are not good enough for production grade systems

Reply:
ACK, Fixed.